### ISSUE TITLE
Incorrect coding of x_start/y_start for space view grid definition

### ISSUE THREAD

**Issue by mo-cbat** [SciTools/iris-grib#292]
For grid definition 90 ("space view"), the following definition for the first point of the grid is incorrect:

https://github.com/SciTools/iris-grib/blob/ac658b404b68f367cd2a56d12dd50439cb3975e9/iris_grib/_load_convert.py#L1235-L1236

Xo/Yo is the "coordinate of origin of the sector image" (GRIB documention)
Xp/Yp is the "coordinate of sub-satellite point"

I'll demonstrate the error using a simple example. For a 3x3 grid and an origin (Xo,Yo) of (0,0), let's say (Xp,Yp) is (2,2), i.e. the middle of the grid. (Xp,Yp) are the coordinates of the grid that intersect (Xo,Yo). The "start" coordinate is therefore (-1,-1), but the current code would set it as (-2,-2). A "+1" is therefore needed.

The impact of this change can be seen in the attached example plot. The "incorrect start coordinate" plot shows that the land mass (data taken from a relevant GRIB file) does not align with the coastlines. This is corrected in the 2nd plot by adding 1 to x_start/y_start.

![coords_incorrect](https://user-images.githubusercontent.com/98948167/158822874-fc19f4e4-9143-4e01-9703-18c99d1fd174.png)
![coords_correct](https://user-images.githubusercontent.com/98948167/158822933-138df260-fc39-4980-b0fd-faaf740e88b6.png)

I've made the change in my fork:

https://github.com/mo-cbat/iris-grib/commit/9ce39243e900ee24e8a35a3aff91d452cf98c087

**Comment by lbdreyer** [comment]
I'm struggling to understand your example so I've tried to draw a diagram to check if this is what you mean:


![image](https://user-images.githubusercontent.com/8101163/161293208-03abc31d-942e-4069-a164-aa6dbf8c776d.png)

As I understand there should be two grids: (i) the one that represents the visible full Earth's disk and (ii) the one that represents the sector image. In the above image, the full Earth grid (i) is represented by a 5x5 grid, but the sector image grid (ii) that may be contained in a GRIB file only covers part of the full grid - a smaller 3x3 grid.

The origin of the sector image (Xo,Yo) is at (0,0) (denoted by a yellow circle in the above image.
The Sub satellite location (Xp, Yp) is at (2, 2) (denoted by a yellow cross), which is the centre of the full Earth grid

From, the above diagram, I would have thought a start coordinate of -2 was correct?

**Comment by mo-cbat** [comment]
Thanks very much for that @lbdreyer! Your example is far better than mine :)

I think the key issue here is that the first cell in grid (i) should be (1, 1), whereas it is (0, 0) in grid (ii). At least that is how I understand the contents of the space view GRIB files that EUMETSAT produce (I'm not aware of any other centres that produce such files with which I could check). So when relating grid (ii) to grid (i) you need to add 1.

I will contact EUMETSAT to see if my understanding is correct and comment here again with my reply.

**Comment by mo-cbat** [comment]
@lbdreyer Right, so I've contacted EUMETSAT and we've deduced that the space view GRIB files that they produce have actually got the wrong values set for (Xo, Yo). Currently they are set to (0, 0), but they should be (1, 1). With the correct values there is no mismatch with the coastlines when plotting using Scitools.

So in conclusion, this pull request is wrong and I'll close it, and Iris is great and works perfectly! Keep up the great work :D
